packages/server/src/schema/uuid/abstract-entity/resolvers.ts

```diff
@@ -588,6 +712,65 @@ async function addRevision({
   }
 }
 
+async function getEntity<S extends Model<'AbstractEntity'> | null>(
+  entityId: number,
+  dataSources: { model: ModelDataSource },
+  decoder: t.Type<S, unknown>
+) {
+  const entity = await dataSources.model.serlo.getUuidWithCustomDecoder({
+    id: entityId,
+    decoder,
+  })
+
+  if (entity === null) {
+    throw 'Nothing found for the provided entityId'
+  }
+
+  return entity
+}
+
+async function verifyAutoreviewEntity(
+  entity:
+    | Model<'Applet'>
+    | Model<'Article'>
+    | Model<'Course'>
+    | Model<'Event'>
+    | Model<'ExerciseGroup'>
+    | Model<'Exercise'>
+    | Model<'Video'>,
+  dataSources: { model: ModelDataSource }
+): Promise<boolean> {
+  const taxonomyTermIds = entity.taxonomyTermIds as number[]
+
+  for (const id of taxonomyTermIds) {
+    const isParentAutoreview = await checkAnyParentAutoreview(id, dataSources)
+    if (!isParentAutoreview) return false
+  }
+
+  return true
```

Comment:
```suggestion
  return (await Promise.all(taxonomyTermIds.map(id => checkAnyParentAutoreview(id, dataSources)))).every(x => x)
```

This will do all the checks also in parallel.